test/Tests/src/Data/Json_Spec.enso

```diff
@@ -106,66 +84,47 @@ spec =
             object_json.should_render_itself
 
         Test.specify "should convert arbitrary types to JSON" <|
-            1.to_json.should_equal (Json.Number 1)
-            1.54.to_json.should_equal (Json.Number 1.54)
-            ["foo", "bar", "baz"].to_json.should_equal <|
-                (Json.Array [Json.String "foo", Json.String "bar", Json.String "baz"])
+            1.to_json.should_equal "1"
+            1.54.to_json.should_equal "1.54"
+            ["foo", "bar", "baz"].to_json.should_equal '["foo","bar","baz"]'
             Author.Value "Tolkien" 1892 . to_json . should_equal <|
-                n = Json.String "Tolkien"
-                y = Json.Number 1892
-                t = Json.String "Value"
-                fields = Map.empty . insert "type" t . insert "name" n . insert "year_of_birth" y
-                Json.Object fields
+                '{"type":"Author","constructor":"Value","name":"Tolkien","year_of_birth":1892}'
 
         Test.specify "should render NaN and Infinity to null" <|
-            Number.nan.to_json.to_text . should_equal "null"
-            Number.positive_infinity.to_json.to_text . should_equal "null"
-            Number.negative_infinity.to_json.to_text . should_equal "null"
+            Number.nan.to_json . should_equal "null"
+            Number.positive_infinity.to_json . should_equal "null"
+            Number.negative_infinity.to_json . should_equal "null"
```

Comment:
:disappointed: I hope we'll be able to figure out something for these cases in the future.